### PR TITLE
Block Form Handling

### PR DESCRIPTION
**Pull request by fosterbrereton**
Adds `blockN` form handling to the DWARF processor.

Until now, we haven't found a case in the wild where essential DWARF data was encoded using one of the `blockN` forms. Today such an example arose. This will handle up to 8 bytes of arbitrary block data, and assumes the result is a 64-bit unsigned value.

### CODE REVIEW

File: src/dwarf.cpp [hunk 1]
```diff
@@ -1421,8 +1467,10 @@ void dwarf::implementation::process_all_dies() {
             attribute_sequence attributes;
             std::tie(die, attributes) = abbreviation_to_die(_s.tellg(), process_mode::complete);
 
+#if ORC_FEATURE(TRACY)
```
Comment (fosterbrereton) [inline src/dwarf.cpp:1470]: squelch a warning.

File: src/dwarf.cpp [hunk 2]
```diff
@@ -1109,6 +1109,57 @@ attribute_value dwarf::implementation::process_form(const attribute& attr,
...
+        std::uint64_t value(0);
+        while (size--) {
+            value <<= 8;
+            value |= read8();
+        }
+
+        result.uint(value);
+    };
+
+    const auto maybe_handle_block = [&](block_type type){
```
Comment (fosterbrereton) [inline src/dwarf.cpp:1142]: The first level of block handling - if the attribute is nonessential, we pass over it like we were doing before. If it is essential, depending on the `blockN` form, we read some number of bytes to discover how much data this block holds. We then forward that size on to `handle_block`, above.

File: src/dwarf.cpp [hunk 3]
```diff
@@ -1109,6 +1109,57 @@ attribute_value dwarf::implementation::process_form(const attribute& attr,
...
+    };
+
+    enum class block_type {
+        one,
+        two,
+        four,
+        uleb,
+    };
+
+    const auto handle_block = [&](std::size_t size) {
```
Comment (fosterbrereton) [inline src/dwarf.cpp:1128]: Where the handling of an essential block takes place. We get a `size` amount from `maybe_handle_block` telling us how many bytes are in this block that we need to process. We read them one at a time, accumulating them in an unsigned 64-bit value. This assumes the value is both an integer, and will fit in 64 bits. If either of this is found to be false, we'll need to revisit this.

**Review by leethomason — APPROVED** [review]
Looks good & good find.

**Pull request merged**